e2e/authentication/login.spec.js

```diff
@@ -0,0 +1,147 @@
+  });
+
+  test.afterEach(async ({ page }) => {
+    await page.request.fetch('/api/database/dump', {
+      method: 'POST',
+    });
+  });
+
+  test.describe('Successful login', () => {
+    test('A user should be able to log in', async ({ page }) => {
+      await page.getByLabel('Email*', { exact: true }).fill('test@testing.com');
+      await page.getByLabel('Password*', { exact: true }).fill('myTestPassw0rd');
+
+      await page.getByRole('button', { name: 'Login' }).click();
+      await expect(page).toHaveTitle('Homepage');
+    });
+
+    test('A user should be able to log in without making his authentication persistent', async ({
+      page,
+      context,
+    }) => {
+      await page.getByLabel('Email*', { exact: true }).fill('test@testing.com');
+      await page.getByLabel('Password*', { exact: true }).fill('myTestPassw0rd');
+
+      await page.getByRole('button', { name: 'Login' }).click();
+      await expect(page).toHaveTitle('Homepage');
+
+      await page.close();
+
+      page = await context.newPage();
+      await page.goto('/admin');
+      await expect(page).toHaveTitle('Strapi Admin');
+    });
+
+    test('A user should be able to log in and make his authentication persistent', async ({
+      page,
+      context,
+    }) => {
+      await page.getByLabel('Email*', { exact: true }).fill('test@testing.com');
+      await page.getByLabel('Password*', { exact: true }).fill('myTestPassw0rd');
+
+      await page.getByLabel('Remember me').click();
+
+      await page.getByRole('button', { name: 'Login' }).click();
+      await expect(page).toHaveTitle('Homepage');
+
+      await page.close();
+
+      page = await context.newPage();
+      await page.goto('/admin');
+      await expect(page).toHaveTitle('Homepage');
+    });
+  });
+
+  test.describe('Rate limit on login', () => {
+    test('Should display a rate limit error message after 5 attempts to login', async ({
+      page,
+      browserName,
+    }) => {
+      await page.request.fetch('/api/config/ratelimit/enable', {
+        method: 'POST',
+        data: { value: true },
+      });
+
+      await page.getByLabel('Email*', { exact: true }).fill(`test@${browserName}.com`);
+      await page.getByLabel('Password*', { exact: true }).fill('wrongPassword');
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+      await page.getByRole('button', { name: 'Login' }).click();
+
+      await expect(page.getByText('Too many requests, please try again later.')).toBeVisible();
+
+      await page.request.fetch('/api/config/ratelimit/enable', {
+        method: 'POST',
+        data: { value: false },
+      });
+    });
+  });
+
+  test.describe('Validation checks', () => {
```

Comment:
These are very unit testy and we know from experience e2e tests take a long time to run. Could they just be in one giant test?

Reply:
I agree, done [here](https://github.com/strapi/strapi/pull/15654/commits/165c30139e489e33749bf388f95477bfda6154c3)